backend/src/mirrors_qa_backend/main.py

```diff
@@ -14,9 +15,16 @@ async def lifespan(_: FastAPI):
 
 
 def create_app(*, debug: bool = True):
-    app = FastAPI(debug=debug, docs_url="/", lifespan=lifespan)
+    app = FastAPI(
+        debug=debug,
+        docs_url="/",
+        lifespan=lifespan,
+        dependencies=[Depends(verify_authorization_header)],
```

Comment:
Actually, the dependency doesn't block requests from unauthenticated users. It just makes sure that if the Authorization header is set, it is of the right format and it is still valid. If the header isn't set, it allows the requests to go through. It's return value is what is actually used by protected endpoints. In a way, it's like Django's `get_current_user`. It can return None or the actual claim. A protected route would ensure it's value is not None. 

I suppose I can refactor it and put them all in the `require_authenticated_worker` dependency.

Reply:
Yes, because with your current implementation, it's not self-documenting.
An explicit approach seems better: the dependency never returns None and is only applied on auth-requiring endpoints